// ant-design-x-vue@1.2.7/es/_util/cssinjs/index.mjs
import { u as r } from "./hooks/useCacheToken.tsx2.mjs";
import { default as W } from "./hooks/useCSSVarRegister.mjs";
import { e, u as o } from "./hooks/useStyleRegister/index.tsx2.mjs";
import t from "./Keyframes.mjs";
import m from "./linters/legacyNotSelectorLinter.mjs";
import a from "./linters/logicalPropertiesLinter.mjs";
import s from "./linters/parentSelectorLinter.mjs";
import { StyleProvider as i, useStyleProvider as p, useStyleInject as n, createCache as f } from "./StyleContext.mjs";
import l from "./theme/createTheme.mjs";
import c from "./theme/Theme.mjs";
import "./theme/ThemeCache.mjs";
import S from "./transformers/legacyLogicalProperties.mjs";
import u from "./transformers/px2rem.mjs";
import { supportWhere as y, supportLogicProps as g } from "./util.mjs";
import { token2CSSVar as b, unit as q } from "./util.mjs";
const R = {
  Theme: c,
  createTheme: l,
  useStyleRegister: o,
  useCacheToken: r,
  createCache: f,
  useStyleInject: n,
  useStyleProvider: p,
  Keyframes: t,
  extractStyle: e,
  // Transformer
  legacyLogicalPropertiesTransformer: S,
  px2remTransformer: u,
  // Linters
  logicalPropertiesLinter: a,
  legacyNotSelectorLinter: m,
  parentSelectorLinter: s,
  // cssinjs
  StyleProvider: i
}, V = {
  supportModernCSS: () => y() && g()
};
export {
  t as Keyframes,
  i as StyleProvider,
  c as Theme,
  V as _experimental,
  f as createCache,
  l as createTheme,
  R as default,
  e as extractStyle,
  S as legacyLogicalPropertiesTransformer,
  m as legacyNotSelectorLinter,
  a as logicalPropertiesLinter,
  s as parentSelectorLinter,
  u as px2remTransformer,
  b as token2CSSVar,
  q as unit,
  W as useCSSVarRegister,
  r as useCacheToken,
  n as useStyleInject,
  p as useStyleProvider,
  o as useStyleRegister
};
