// ant-design-x-vue@1.2.7/es/attachments/FileList/FileList2.mjs
import { inject as ft, ref as E, h as Q, getCurrentInstance as ut, nextTick as Te, reactive as st, defineComponent as ze, createVNode as c, computed as dt, useTemplateRef as mt, watchEffect as gt, onWatcherCleanup as pt, watch as bt, Fragment as vt } from "vue";
import { c as J } from "../Attachments2.mjs";
import { Button as _ } from "ant-design-vue";
import I from "../../_util/hooks/use-state.mjs";
import { useAttachmentContextInject as ht } from "../context.mjs";
import yt from "../SilentUploader.mjs";
import Ot from "./FileListCard.mjs";
function g(e, r) {
  wt(e) && (e = "100%");
  var t = Pt(e);
  return e = r === 360 ? e : Math.min(r, Math.max(0, parseFloat(e))), t && (e = parseInt(String(e * r), 10) / 100), Math.abs(e - r) < 1e-6 ? 1 : (r === 360 ? e = (e < 0 ? e % r + r : e % r) / parseFloat(String(r)) : e = e % r / parseFloat(String(r)), e);
}
function wt(e) {
  return typeof e == "string" && e.indexOf(".") !== -1 && parseFloat(e) === 1;
}
function Pt(e) {
  return typeof e == "string" && e.indexOf("%") !== -1;
}
function Ct(e) {
  return e = parseFloat(e), (isNaN(e) || e < 0 || e > 1) && (e = 1), e;
}
function F(e) {
  return e <= 1 ? "".concat(Number(e) * 100, "%") : e;
}
function N(e) {
  return e.length === 1 ? "0" + e : String(e);
}
function St(e, r, t) {
  return {
    r: g(e, 255) * 255,
    g: g(r, 255) * 255,
    b: g(t, 255) * 255
  };
}
function V(e, r, t) {
  return t < 0 && (t += 1), t > 1 && (t -= 1), t < 1 / 6 ? e + (r - e) * (6 * t) : t < 1 / 2 ? r : t < 2 / 3 ? e + (r - e) * (2 / 3 - t) * 6 : e;
}
function jt(e, r, t) {
  var n, a, i;
  if (e = g(e, 360), r = g(r, 100), t = g(t, 100), r === 0)
    a = t, i = t, n = t;
  else {
    var o = t < 0.5 ? t * (1 + r) : t + r - t * r, l = 2 * t - o;
    n = V(l, o, e + 1 / 3), a = V(l, o, e), i = V(l, o, e - 1 / 3);
  }
  return { r: n * 255, g: a * 255, b: i * 255 };
}
function xt(e, r, t) {
  e = g(e, 255), r = g(r, 255), t = g(t, 255);
  var n = Math.max(e, r, t), a = Math.min(e, r, t), i = 0, o = n, l = n - a, f = n === 0 ? 0 : l / n;
  if (n === a)
    i = 0;
  else {
    switch (n) {
      case e:
        i = (r - t) / l + (r < t ? 6 : 0);
        break;
      case r:
        i = (t - e) / l + 2;
        break;
      case t:
        i = (e - r) / l + 4;
        break;
    }
    i /= 6;
  }
  return { h: i, s: f, v: o };
}
function Ft(e, r, t) {
  e = g(e, 360) * 6, r = g(r, 100), t = g(t, 100);
  var n = Math.floor(e), a = e - n, i = t * (1 - r), o = t * (1 - a * r), l = t * (1 - (1 - a) * r), f = n % 6, d = [t, o, i, i, l, t][f], b = [l, t, t, o, i, i][f], v = [i, i, l, t, t, o][f];
  return { r: d * 255, g: b * 255, b: v * 255 };
}
function At(e, r, t, n) {
  var a = [
    N(Math.round(e).toString(16)),
    N(Math.round(r).toString(16)),
    N(Math.round(t).toString(16))
  ];
  return a.join("");
}
function K(e) {
  return m(e) / 255;
}
function m(e) {
  return parseInt(e, 16);
}
var ee = {
  aliceblue: "#f0f8ff",
  antiquewhite: "#faebd7",
  aqua: "#00ffff",
  aquamarine: "#7fffd4",
  azure: "#f0ffff",
  beige: "#f5f5dc",
  bisque: "#ffe4c4",
  black: "#000000",
  blanchedalmond: "#ffebcd",
  blue: "#0000ff",
  blueviolet: "#8a2be2",
  brown: "#a52a2a",
  burlywood: "#deb887",
  cadetblue: "#5f9ea0",
  chartreuse: "#7fff00",
  chocolate: "#d2691e",
  coral: "#ff7f50",
  cornflowerblue: "#6495ed",
  cornsilk: "#fff8dc",
  crimson: "#dc143c",
  cyan: "#00ffff",
  darkblue: "#00008b",
  darkcyan: "#008b8b",
  darkgoldenrod: "#b8860b",
  darkgray: "#a9a9a9",
  darkgreen: "#006400",
  darkgrey: "#a9a9a9",
  darkkhaki: "#bdb76b",
  darkmagenta: "#8b008b",
  darkolivegreen: "#556b2f",
  darkorange: "#ff8c00",
  darkorchid: "#9932cc",
  darkred: "#8b0000",
  darksalmon: "#e9967a",
  darkseagreen: "#8fbc8f",
  darkslateblue: "#483d8b",
  darkslategray: "#2f4f4f",
  darkslategrey: "#2f4f4f",
  darkturquoise: "#00ced1",
  darkviolet: "#9400d3",
  deeppink: "#ff1493",
  deepskyblue: "#00bfff",
  dimgray: "#696969",
  dimgrey: "#696969",
  dodgerblue: "#1e90ff",
  firebrick: "#b22222",
  floralwhite: "#fffaf0",
  forestgreen: "#228b22",
  fuchsia: "#ff00ff",
  gainsboro: "#dcdcdc",
  ghostwhite: "#f8f8ff",
  goldenrod: "#daa520",
  gold: "#ffd700",
  gray: "#808080",
  green: "#008000",
  greenyellow: "#adff2f",
  grey: "#808080",
  honeydew: "#f0fff0",
  hotpink: "#ff69b4",
  indianred: "#cd5c5c",
  indigo: "#4b0082",
  ivory: "#fffff0",
  khaki: "#f0e68c",
  lavenderblush: "#fff0f5",
  lavender: "#e6e6fa",
  lawngreen: "#7cfc00",
  lemonchiffon: "#fffacd",
  lightblue: "#add8e6",
  lightcoral: "#f08080",
  lightcyan: "#e0ffff",
  lightgoldenrodyellow: "#fafad2",
  lightgray: "#d3d3d3",
  lightgreen: "#90ee90",
  lightgrey: "#d3d3d3",
  lightpink: "#ffb6c1",
  lightsalmon: "#ffa07a",
  lightseagreen: "#20b2aa",
  lightskyblue: "#87cefa",
  lightslategray: "#778899",
  lightslategrey: "#778899",
  lightsteelblue: "#b0c4de",
  lightyellow: "#ffffe0",
  lime: "#00ff00",
  limegreen: "#32cd32",
  linen: "#faf0e6",
  magenta: "#ff00ff",
  maroon: "#800000",
  mediumaquamarine: "#66cdaa",
  mediumblue: "#0000cd",
  mediumorchid: "#ba55d3",
  mediumpurple: "#9370db",
  mediumseagreen: "#3cb371",
  mediumslateblue: "#7b68ee",
  mediumspringgreen: "#00fa9a",
  mediumturquoise: "#48d1cc",
  mediumvioletred: "#c71585",
  midnightblue: "#191970",
  mintcream: "#f5fffa",
  mistyrose: "#ffe4e1",
  moccasin: "#ffe4b5",
  navajowhite: "#ffdead",
  navy: "#000080",
  oldlace: "#fdf5e6",
  olive: "#808000",
  olivedrab: "#6b8e23",
  orange: "#ffa500",
  orangered: "#ff4500",
  orchid: "#da70d6",
  palegoldenrod: "#eee8aa",
  palegreen: "#98fb98",
  paleturquoise: "#afeeee",
  palevioletred: "#db7093",
  papayawhip: "#ffefd5",
  peachpuff: "#ffdab9",
  peru: "#cd853f",
  pink: "#ffc0cb",
  plum: "#dda0dd",
  powderblue: "#b0e0e6",
  purple: "#800080",
  rebeccapurple: "#663399",
  red: "#ff0000",
  rosybrown: "#bc8f8f",
  royalblue: "#4169e1",
  saddlebrown: "#8b4513",
  salmon: "#fa8072",
  sandybrown: "#f4a460",
  seagreen: "#2e8b57",
  seashell: "#fff5ee",
  sienna: "#a0522d",
  silver: "#c0c0c0",
  skyblue: "#87ceeb",
  slateblue: "#6a5acd",
  slategray: "#708090",
  slategrey: "#708090",
  snow: "#fffafa",
  springgreen: "#00ff7f",
  steelblue: "#4682b4",
  tan: "#d2b48c",
  teal: "#008080",
  thistle: "#d8bfd8",
  tomato: "#ff6347",
  turquoise: "#40e0d0",
  violet: "#ee82ee",
  wheat: "#f5deb3",
  white: "#ffffff",
  whitesmoke: "#f5f5f5",
  yellow: "#ffff00",
  yellowgreen: "#9acd32"
};
function S(e) {
  var r = { r: 0, g: 0, b: 0 }, t = 1, n = null, a = null, i = null, o = !1, l = !1;
  return typeof e == "string" && (e = Tt(e)), typeof e == "object" && (h(e.r) && h(e.g) && h(e.b) ? (r = St(e.r, e.g, e.b), o = !0, l = String(e.r).substr(-1) === "%" ? "prgb" : "rgb") : h(e.h) && h(e.s) && h(e.v) ? (n = F(e.s), a = F(e.v), r = Ft(e.h, n, a), o = !0, l = "hsv") : h(e.h) && h(e.s) && h(e.l) && (n = F(e.s), i = F(e.l), r = jt(e.h, n, i), o = !0, l = "hsl"), Object.prototype.hasOwnProperty.call(e, "a") && (t = e.a)), t = Ct(t), {
    ok: o,
    format: e.format || l,
    r: Math.min(255, Math.max(r.r, 0)),
    g: Math.min(255, Math.max(r.g, 0)),
    b: Math.min(255, Math.max(r.b, 0)),
    a: t
  };
}
var $t = "[-\\+]?\\d+%?", Mt = "[-\\+]?\\d*\\.\\d+%?", y = "(?:".concat(Mt, ")|(?:").concat($t, ")"), L = "[\\s|\\(]+(".concat(y, ")[,|\\s]+(").concat(y, ")[,|\\s]+(").concat(y, ")\\s*\\)?"), H = "[\\s|\\(]+(".concat(y, ")[,|\\s]+(").concat(y, ")[,|\\s]+(").concat(y, ")[,|\\s]+(").concat(y, ")\\s*\\)?"), p = {
  CSS_UNIT: new RegExp(y),
  rgb: new RegExp("rgb" + L),
  rgba: new RegExp("rgba" + H),
  hsl: new RegExp("hsl" + L),
  hsla: new RegExp("hsla" + H),
  hsv: new RegExp("hsv" + L),
  hsva: new RegExp("hsva" + H),
  hex3: /^#?([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})$/,
  hex6: /^#?([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})$/,
  hex4: /^#?([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})([0-9a-fA-F]{1})$/,
  hex8: /^#?([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})([0-9a-fA-F]{2})$/
};
function Tt(e) {
  if (e = e.trim().toLowerCase(), e.length === 0)
    return !1;
  var r = !1;
  if (ee[e])
    e = ee[e], r = !0;
  else if (e === "transparent")
    return { r: 0, g: 0, b: 0, a: 0, format: "name" };
  var t = p.rgb.exec(e);
  return t ? { r: t[1], g: t[2], b: t[3] } : (t = p.rgba.exec(e), t ? { r: t[1], g: t[2], b: t[3], a: t[4] } : (t = p.hsl.exec(e), t ? { h: t[1], s: t[2], l: t[3] } : (t = p.hsla.exec(e), t ? { h: t[1], s: t[2], l: t[3], a: t[4] } : (t = p.hsv.exec(e), t ? { h: t[1], s: t[2], v: t[3] } : (t = p.hsva.exec(e), t ? { h: t[1], s: t[2], v: t[3], a: t[4] } : (t = p.hex8.exec(e), t ? {
    r: m(t[1]),
    g: m(t[2]),
    b: m(t[3]),
    a: K(t[4]),
    format: r ? "name" : "hex8"
  } : (t = p.hex6.exec(e), t ? {
    r: m(t[1]),
    g: m(t[2]),
    b: m(t[3]),
    format: r ? "name" : "hex"
  } : (t = p.hex4.exec(e), t ? {
    r: m(t[1] + t[1]),
    g: m(t[2] + t[2]),
    b: m(t[3] + t[3]),
    a: K(t[4] + t[4]),
    format: r ? "name" : "hex8"
  } : (t = p.hex3.exec(e), t ? {
    r: m(t[1] + t[1]),
    g: m(t[2] + t[2]),
    b: m(t[3] + t[3]),
    format: r ? "name" : "hex"
  } : !1)))))))));
}
function h(e) {
  return !!p.CSS_UNIT.exec(String(e));
}
var A = 2, te = 0.16, zt = 0.05, Et = 0.05, _t = 0.15, Ee = 5, _e = 4, It = [{
  index: 7,
  opacity: 0.15
}, {
  index: 6,
  opacity: 0.25
}, {
  index: 5,
  opacity: 0.3
}, {
  index: 5,
  opacity: 0.45
}, {
  index: 5,
  opacity: 0.65
}, {
  index: 5,
  opacity: 0.85
}, {
  index: 4,
  opacity: 0.9
}, {
  index: 3,
  opacity: 0.95
}, {
  index: 2,
  opacity: 0.97
}, {
  index: 1,
  opacity: 0.98
}];
function re(e) {
  var r = e.r, t = e.g, n = e.b, a = xt(r, t, n);
  return {
    h: a.h * 360,
    s: a.s,
    v: a.v
  };
}
function $(e) {
  var r = e.r, t = e.g, n = e.b;
  return "#".concat(At(r, t, n));
}
function Nt(e, r, t) {
  var n = t / 100, a = {
    r: (r.r - e.r) * n + e.r,
    g: (r.g - e.g) * n + e.g,
    b: (r.b - e.b) * n + e.b
  };
  return a;
}
function ne(e, r, t) {
  var n;
  return Math.round(e.h) >= 60 && Math.round(e.h) <= 240 ? n = t ? Math.round(e.h) - A * r : Math.round(e.h) + A * r : n = t ? Math.round(e.h) + A * r : Math.round(e.h) - A * r, n < 0 ? n += 360 : n >= 360 && (n -= 360), n;
}
function ae(e, r, t) {
  if (e.h === 0 && e.s === 0)
    return e.s;
  var n;
  return t ? n = e.s - te * r : r === _e ? n = e.s + te : n = e.s + zt * r, n > 1 && (n = 1), t && r === Ee && n > 0.1 && (n = 0.1), n < 0.06 && (n = 0.06), Number(n.toFixed(2));
}
function ie(e, r, t) {
  var n;
  return t ? n = e.v + Et * r : n = e.v - _t * r, n > 1 && (n = 1), Number(n.toFixed(2));
}
function B(e) {
  for (var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, t = [], n = S(e), a = Ee; a > 0; a -= 1) {
    var i = re(n), o = $(S({
      h: ne(i, a, !0),
      s: ae(i, a, !0),
      v: ie(i, a, !0)
    }));
    t.push(o);
  }
  t.push($(n));
  for (var l = 1; l <= _e; l += 1) {
    var f = re(n), d = $(S({
      h: ne(f, l),
      s: ae(f, l),
      v: ie(f, l)
    }));
    t.push(d);
  }
  return r.theme === "dark" ? It.map(function(b) {
    var v = b.index, C = b.opacity, P = $(Nt(S(r.backgroundColor || "#141414"), S(t[v]), C * 100));
    return P;
  }) : t;
}
var R = {
  red: "#F5222D",
  volcano: "#FA541C",
  orange: "#FA8C16",
  gold: "#FAAD14",
  yellow: "#FADB14",
  lime: "#A0D911",
  green: "#52C41A",
  cyan: "#13C2C2",
  blue: "#1890FF",
  geekblue: "#2F54EB",
  purple: "#722ED1",
  magenta: "#EB2F96",
  grey: "#666666"
}, M = {}, k = {};
Object.keys(R).forEach(function(e) {
  M[e] = B(R[e]), M[e].primary = M[e][5], k[e] = B(R[e], {
    theme: "dark",
    backgroundColor: "#141414"
  }), k[e].primary = k[e][5];
});
var Vt = M.blue, Lt = Symbol("iconContext"), Ie = function() {
  return ft(Lt, {
    prefixCls: E("anticon"),
    rootClassName: E(""),
    csp: E()
  });
};
function W() {
  return !!(typeof window < "u" && window.document && window.document.createElement);
}
function Ht(e, r) {
  return e && e.contains ? e.contains(r) : !1;
}
var le = "data-vc-order", Rt = "vc-icon-key", D = /* @__PURE__ */ new Map();
function Ne() {
  var e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, r = e.mark;
  return r ? r.startsWith("data-") ? r : "data-".concat(r) : Rt;
}
function U(e) {
  if (e.attachTo)
    return e.attachTo;
  var r = document.querySelector("head");
  return r || document.body;
}
function kt(e) {
  return e === "queue" ? "prependQueue" : e ? "prepend" : "append";
}
function Ve(e) {
  return Array.from((D.get(e) || e).children).filter(function(r) {
    return r.tagName === "STYLE";
  });
}
function Le(e) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
  if (!W())
    return null;
  var t = r.csp, n = r.prepend, a = document.createElement("style");
  a.setAttribute(le, kt(n)), t && t.nonce && (a.nonce = t.nonce), a.innerHTML = e;
  var i = U(r), o = i.firstChild;
  if (n) {
    if (n === "queue") {
      var l = Ve(i).filter(function(f) {
        return ["prepend", "prependQueue"].includes(f.getAttribute(le));
      });
      if (l.length)
        return i.insertBefore(a, l[l.length - 1].nextSibling), a;
    }
    i.insertBefore(a, o);
  } else
    i.appendChild(a);
  return a;
}
function Bt(e) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, t = U(r);
  return Ve(t).find(function(n) {
    return n.getAttribute(Ne(r)) === e;
  });
}
function Dt(e, r) {
  var t = D.get(e);
  if (!t || !Ht(document, t)) {
    var n = Le("", r), a = n.parentNode;
    D.set(e, a), e.removeChild(n);
  }
}
function qt(e, r) {
  var t = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : {}, n = U(t);
  Dt(n, t);
  var a = Bt(r, t);
  if (a)
    return t.csp && t.csp.nonce && a.nonce !== t.csp.nonce && (a.nonce = t.csp.nonce), a.innerHTML !== e && (a.innerHTML = e), a;
  var i = Le(e, t);
  return i.setAttribute(Ne(t), r), i;
}
function oe(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Wt(e, a, t[a]);
    });
  }
  return e;
}
function Wt(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
function ce(e) {
  return typeof e == "object" && typeof e.name == "string" && typeof e.theme == "string" && (typeof e.icon == "object" || typeof e.icon == "function");
}
function q(e, r, t) {
  return t ? Q(e.tag, oe({
    key: r
  }, t, e.attrs), (e.children || []).map(function(n, a) {
    return q(n, "".concat(r, "-").concat(e.tag, "-").concat(a));
  })) : Q(e.tag, oe({
    key: r
  }, e.attrs), (e.children || []).map(function(n, a) {
    return q(n, "".concat(r, "-").concat(e.tag, "-").concat(a));
  }));
}
function He(e) {
  return B(e)[0];
}
function Re(e) {
  return e ? Array.isArray(e) ? e : [e] : [];
}
var Ut = `
.anticon {
  display: inline-block;
  color: inherit;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.anticon > * {
  line-height: 1;
}

.anticon svg {
  display: inline-block;
}

.anticon::before {
  display: none;
}

.anticon .anticon-icon {
  display: block;
}

.anticon[tabindex] {
  cursor: pointer;
}

.anticon-spin::before,
.anticon-spin {
  display: inline-block;
  -webkit-animation: loadingCircle 1s infinite linear;
  animation: loadingCircle 1s infinite linear;
}

@-webkit-keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
`;
function ke(e) {
  return e && e.getRootNode && e.getRootNode();
}
function Zt(e) {
  return W() ? ke(e) instanceof ShadowRoot : !1;
}
function Xt(e) {
  return Zt(e) ? ke(e) : null;
}
var Yt = function() {
  var r = Ie(), t = r.prefixCls, n = r.csp, a = ut(), i = Ut;
  t && (i = i.replace(/anticon/g, t.value)), Te(function() {
    if (W()) {
      var o = a.vnode.el, l = Xt(o);
      qt(i, "@ant-design-vue-icons", {
        prepend: !0,
        csp: n.value,
        attachTo: l
      });
    }
  });
}, Gt = ["icon", "primaryColor", "secondaryColor"];
function Qt(e, r) {
  if (e == null) return {};
  var t = Jt(e, r), n, a;
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (a = 0; a < i.length; a++)
      n = i[a], !(r.indexOf(n) >= 0) && Object.prototype.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
  }
  return t;
}
function Jt(e, r) {
  if (e == null) return {};
  var t = {}, n = Object.keys(e), a, i;
  for (i = 0; i < n.length; i++)
    a = n[i], !(r.indexOf(a) >= 0) && (t[a] = e[a]);
  return t;
}
function T(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Kt(e, a, t[a]);
    });
  }
  return e;
}
function Kt(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var x = st({
  primaryColor: "#333",
  secondaryColor: "#E6E6E6",
  calculated: !1
});
function er(e) {
  var r = e.primaryColor, t = e.secondaryColor;
  x.primaryColor = r, x.secondaryColor = t || He(r), x.calculated = !!t;
}
function tr() {
  return T({}, x);
}
var O = function(r, t) {
  var n = T({}, r, t.attrs), a = n.icon, i = n.primaryColor, o = n.secondaryColor, l = Qt(n, Gt), f = x;
  if (i && (f = {
    primaryColor: i,
    secondaryColor: o || He(i)
  }), ce(a), !ce(a))
    return null;
  var d = a;
  return d && typeof d.icon == "function" && (d = T({}, d, {
    icon: d.icon(f.primaryColor, f.secondaryColor)
  })), q(d.icon, "svg-".concat(d.name), T({}, l, {
    "data-icon": d.name,
    width: "1em",
    height: "1em",
    fill: "currentColor",
    "aria-hidden": "true"
  }));
};
O.props = {
  icon: Object,
  primaryColor: String,
  secondaryColor: String,
  focusable: String
};
O.inheritAttrs = !1;
O.displayName = "IconBase";
O.getTwoToneColors = tr;
O.setTwoToneColors = er;
function rr(e, r) {
  return lr(e) || ir(e, r) || ar(e, r) || nr();
}
function nr() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function ar(e, r) {
  if (e) {
    if (typeof e == "string") return fe(e, r);
    var t = Object.prototype.toString.call(e).slice(8, -1);
    if (t === "Object" && e.constructor && (t = e.constructor.name), t === "Map" || t === "Set") return Array.from(e);
    if (t === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t)) return fe(e, r);
  }
}
function fe(e, r) {
  (r == null || r > e.length) && (r = e.length);
  for (var t = 0, n = new Array(r); t < r; t++)
    n[t] = e[t];
  return n;
}
function ir(e, r) {
  var t = e == null ? null : typeof Symbol < "u" && e[Symbol.iterator] || e["@@iterator"];
  if (t != null) {
    var n = [], a = !0, i = !1, o, l;
    try {
      for (t = t.call(e); !(a = (o = t.next()).done) && (n.push(o.value), !(r && n.length === r)); a = !0)
        ;
    } catch (f) {
      i = !0, l = f;
    } finally {
      try {
        !a && t.return != null && t.return();
      } finally {
        if (i) throw l;
      }
    }
    return n;
  }
}
function lr(e) {
  if (Array.isArray(e)) return e;
}
function Be(e) {
  var r = Re(e), t = rr(r, 2), n = t[0], a = t[1];
  return O.setTwoToneColors({
    primaryColor: n,
    secondaryColor: a
  });
}
function or() {
  var e = O.getTwoToneColors();
  return e.calculated ? [e.primaryColor, e.secondaryColor] : e.primaryColor;
}
var cr = ze({
  name: "InsertStyles",
  setup: function() {
    return Yt(), function() {
      return null;
    };
  }
}), fr = ["class", "icon", "spin", "rotate", "tabindex", "twoToneColor", "onClick"];
function ur(e, r) {
  return gr(e) || mr(e, r) || dr(e, r) || sr();
}
function sr() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function dr(e, r) {
  if (e) {
    if (typeof e == "string") return ue(e, r);
    var t = Object.prototype.toString.call(e).slice(8, -1);
    if (t === "Object" && e.constructor && (t = e.constructor.name), t === "Map" || t === "Set") return Array.from(e);
    if (t === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t)) return ue(e, r);
  }
}
function ue(e, r) {
  (r == null || r > e.length) && (r = e.length);
  for (var t = 0, n = new Array(r); t < r; t++)
    n[t] = e[t];
  return n;
}
function mr(e, r) {
  var t = e == null ? null : typeof Symbol < "u" && e[Symbol.iterator] || e["@@iterator"];
  if (t != null) {
    var n = [], a = !0, i = !1, o, l;
    try {
      for (t = t.call(e); !(a = (o = t.next()).done) && (n.push(o.value), !(r && n.length === r)); a = !0)
        ;
    } catch (f) {
      i = !0, l = f;
    } finally {
      try {
        !a && t.return != null && t.return();
      } finally {
        if (i) throw l;
      }
    }
    return n;
  }
}
function gr(e) {
  if (Array.isArray(e)) return e;
}
function se(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      j(e, a, t[a]);
    });
  }
  return e;
}
function j(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
function pr(e, r) {
  if (e == null) return {};
  var t = br(e, r), n, a;
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (a = 0; a < i.length; a++)
      n = i[a], !(r.indexOf(n) >= 0) && Object.prototype.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
  }
  return t;
}
function br(e, r) {
  if (e == null) return {};
  var t = {}, n = Object.keys(e), a, i;
  for (i = 0; i < n.length; i++)
    a = n[i], !(r.indexOf(a) >= 0) && (t[a] = e[a]);
  return t;
}
Be(Vt.primary);
var u = function(r, t) {
  var n, a = se({}, r, t.attrs), i = a.class, o = a.icon, l = a.spin, f = a.rotate, d = a.tabindex, b = a.twoToneColor, v = a.onClick, C = pr(a, fr), P = Ie(), s = P.prefixCls, w = P.rootClassName, at = (n = {}, j(n, w.value, !!w.value), j(n, s.value, !0), j(n, "".concat(s.value, "-").concat(o.name), !!o.name), j(n, "".concat(s.value, "-spin"), !!l || o.name === "loading"), n), z = d;
  z === void 0 && v && (z = -1);
  var it = f ? {
    msTransform: "rotate(".concat(f, "deg)"),
    transform: "rotate(".concat(f, "deg)")
  } : void 0, lt = Re(b), G = ur(lt, 2), ot = G[0], ct = G[1];
  return c("span", se({
    role: "img",
    "aria-label": o.name
  }, C, {
    onClick: v,
    class: [at, i],
    tabindex: z
  }), [c(O, {
    icon: o,
    primaryColor: ot,
    secondaryColor: ct,
    style: it
  }, null), c(cr, null, null)]);
};
u.props = {
  spin: Boolean,
  rotate: Number,
  icon: Object,
  twoToneColor: [String, Array]
};
u.displayName = "AntdIcon";
u.inheritAttrs = !1;
u.getTwoToneColor = or;
u.setTwoToneColor = Be;
var vr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z" } }] }, name: "arrow-up", theme: "outlined" };
function de(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      hr(e, a, t[a]);
    });
  }
  return e;
}
function hr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var De = function(r, t) {
  var n = de({}, r, t.attrs);
  return c(u, de({}, n, {
    icon: vr
  }), null);
};
De.displayName = "ArrowUpOutlined";
De.inheritAttrs = !1;
var yr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "defs", attrs: {}, children: [{ tag: "style", attrs: {} }] }, { tag: "path", attrs: { d: "M682 455V311l-76 76v68c-.1 50.7-42 92.1-94 92a95.8 95.8 0 01-52-15l-54 55c29.1 22.4 65.9 36 106 36 93.8 0 170-75.1 170-168z" } }, { tag: "path", attrs: { d: "M833 446h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254-63 0-120.7-23-165-61l-54 54a334.01 334.01 0 00179 81v102H326c-13.9 0-24.9 14.3-25 32v36c.1 4.4 2.9 8 6 8h408c3.2 0 6-3.6 6-8v-36c0-17.7-11-32-25-32H547V782c165.3-17.9 294-157.9 294-328 0-4.4-3.6-8-8-8zm13.1-377.7l-43.5-41.9a8 8 0 00-11.2.1l-129 129C634.3 101.2 577 64 511 64c-93.9 0-170 75.3-170 168v224c0 6.7.4 13.3 1.2 19.8l-68 68A252.33 252.33 0 01258 454c-.2-4.4-3.8-8-8-8h-60c-4.4 0-8 3.6-8 8 0 53 12.5 103 34.6 147.4l-137 137a8.03 8.03 0 000 11.3l42.7 42.7c3.1 3.1 8.2 3.1 11.3 0L846.2 79.8l.1-.1c3.1-3.2 3-8.3-.2-11.4zM417 401V232c0-50.6 41.9-92 94-92 46 0 84.1 32.3 92.3 74.7L417 401z" } }] }, name: "audio-muted", theme: "outlined" };
function me(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Or(e, a, t[a]);
    });
  }
  return e;
}
function Or(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var qe = function(r, t) {
  var n = me({}, r, t.attrs);
  return c(u, me({}, n, {
    icon: yr
  }), null);
};
qe.displayName = "AudioMutedOutlined";
qe.inheritAttrs = !1;
var wr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M842 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8 0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1zM512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm-94-392c0-50.6 41.9-92 94-92s94 41.4 94 92v224c0 50.6-41.9 92-94 92s-94-41.4-94-92V232z" } }] }, name: "audio", theme: "outlined" };
function ge(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Pr(e, a, t[a]);
    });
  }
  return e;
}
function Pr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var We = function(r, t) {
  var n = ge({}, r, t.attrs);
  return c(u, ge({}, n, {
    icon: wr
  }), null);
};
We.displayName = "AudioOutlined";
We.inheritAttrs = !1;
var Cr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "defs", attrs: {}, children: [{ tag: "style", attrs: {} }] }, { tag: "path", attrs: { d: "M899.1 869.6l-53-305.6H864c14.4 0 26-11.6 26-26V346c0-14.4-11.6-26-26-26H618V138c0-14.4-11.6-26-26-26H432c-14.4 0-26 11.6-26 26v182H160c-14.4 0-26 11.6-26 26v192c0 14.4 11.6 26 26 26h17.9l-53 305.6a25.95 25.95 0 0025.6 30.4h723c1.5 0 3-.1 4.4-.4a25.88 25.88 0 0021.2-30zM204 390h272V182h72v208h272v104H204V390zm468 440V674c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v156H416V674c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v156H202.8l45.1-260H776l45.1 260H672z" } }] }, name: "clear", theme: "outlined" };
function pe(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Sr(e, a, t[a]);
    });
  }
  return e;
}
function Sr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Ue = function(r, t) {
  var n = pe({}, r, t.attrs);
  return c(u, pe({}, n, {
    icon: Cr
  }), null);
};
Ue.displayName = "ClearOutlined";
Ue.inheritAttrs = !1;
var jr = { icon: { tag: "svg", attrs: { "fill-rule": "evenodd", viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z" } }] }, name: "close-circle", theme: "filled" };
function be(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      xr(e, a, t[a]);
    });
  }
  return e;
}
function xr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Ze = function(r, t) {
  var n = be({}, r, t.attrs);
  return c(u, be({}, n, {
    icon: jr
  }), null);
};
Ze.displayName = "CloseCircleFilled";
Ze.inheritAttrs = !1;
var Fr = { icon: { tag: "svg", attrs: { "fill-rule": "evenodd", viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z" } }] }, name: "close", theme: "outlined" };
function ve(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Ar(e, a, t[a]);
    });
  }
  return e;
}
function Ar(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Xe = function(r, t) {
  var n = ve({}, r, t.attrs);
  return c(u, ve({}, n, {
    icon: Fr
  }), null);
};
Xe.displayName = "CloseOutlined";
Xe.inheritAttrs = !1;
var $r = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z" } }] }, name: "ellipsis", theme: "outlined" };
function he(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Mr(e, a, t[a]);
    });
  }
  return e;
}
function Mr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Ye = function(r, t) {
  var n = he({}, r, t.attrs);
  return c(u, he({}, n, {
    icon: $r
  }), null);
};
Ye.displayName = "EllipsisOutlined";
Ye.inheritAttrs = !1;
var Tr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM575.34 477.84l-61.22 102.3L452.3 477.8a12 12 0 00-10.27-5.79h-38.44a12 12 0 00-6.4 1.85 12 12 0 00-3.75 16.56l82.34 130.42-83.45 132.78a12 12 0 00-1.84 6.39 12 12 0 0012 12h34.46a12 12 0 0010.21-5.7l62.7-101.47 62.3 101.45a12 12 0 0010.23 5.72h37.48a12 12 0 006.48-1.9 12 12 0 003.62-16.58l-83.83-130.55 85.3-132.47a12 12 0 001.9-6.5 12 12 0 00-12-12h-35.7a12 12 0 00-10.29 5.84z" } }] }, name: "file-excel", theme: "filled" };
function ye(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      zr(e, a, t[a]);
    });
  }
  return e;
}
function zr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Ge = function(r, t) {
  var n = ye({}, r, t.attrs);
  return c(u, ye({}, n, {
    icon: Tr
  }), null);
};
Ge.displayName = "FileExcelFilled";
Ge.inheritAttrs = !1;
var Er = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7L639.4 73.4c-6-6-14.2-9.4-22.7-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.6-9.4-22.6zM400 402c22.1 0 40 17.9 40 40s-17.9 40-40 40-40-17.9-40-40 17.9-40 40-40zm296 294H328c-6.7 0-10.4-7.7-6.3-12.9l99.8-127.2a8 8 0 0112.6 0l41.1 52.4 77.8-99.2a8 8 0 0112.6 0l136.5 174c4.3 5.2.5 12.9-6.1 12.9zm-94-370V137.8L790.2 326H602z" } }] }, name: "file-image", theme: "filled" };
function Oe(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      _r(e, a, t[a]);
    });
  }
  return e;
}
function _r(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Qe = function(r, t) {
  var n = Oe({}, r, t.attrs);
  return c(u, Oe({}, n, {
    icon: Er
  }), null);
};
Qe.displayName = "FileImageFilled";
Qe.inheritAttrs = !1;
var Ir = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM426.13 600.93l59.11 132.97a16 16 0 0014.62 9.5h24.06a16 16 0 0014.63-9.51l59.1-133.35V758a16 16 0 0016.01 16H641a16 16 0 0016-16V486a16 16 0 00-16-16h-34.75a16 16 0 00-14.67 9.62L512.1 662.2l-79.48-182.59a16 16 0 00-14.67-9.61H383a16 16 0 00-16 16v272a16 16 0 0016 16h27.13a16 16 0 0016-16V600.93z" } }] }, name: "file-markdown", theme: "filled" };
function we(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Nr(e, a, t[a]);
    });
  }
  return e;
}
function Nr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Je = function(r, t) {
  var n = we({}, r, t.attrs);
  return c(u, we({}, n, {
    icon: Ir
  }), null);
};
Je.displayName = "FileMarkdownFilled";
Je.inheritAttrs = !1;
var Vr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM633.22 637.26c-15.18-.5-31.32.67-49.65 2.96-24.3-14.99-40.66-35.58-52.28-65.83l1.07-4.38 1.24-5.18c4.3-18.13 6.61-31.36 7.3-44.7.52-10.07-.04-19.36-1.83-27.97-3.3-18.59-16.45-29.46-33.02-30.13-15.45-.63-29.65 8-33.28 21.37-5.91 21.62-2.45 50.07 10.08 98.59-15.96 38.05-37.05 82.66-51.2 107.54-18.89 9.74-33.6 18.6-45.96 28.42-16.3 12.97-26.48 26.3-29.28 40.3-1.36 6.49.69 14.97 5.36 21.92 5.3 7.88 13.28 13 22.85 13.74 24.15 1.87 53.83-23.03 86.6-79.26 3.29-1.1 6.77-2.26 11.02-3.7l11.9-4.02c7.53-2.54 12.99-4.36 18.39-6.11 23.4-7.62 41.1-12.43 57.2-15.17 27.98 14.98 60.32 24.8 82.1 24.8 17.98 0 30.13-9.32 34.52-23.99 3.85-12.88.8-27.82-7.48-36.08-8.56-8.41-24.3-12.43-45.65-13.12zM385.23 765.68v-.36l.13-.34a54.86 54.86 0 015.6-10.76c4.28-6.58 10.17-13.5 17.47-20.87 3.92-3.95 8-7.8 12.79-12.12 1.07-.96 7.91-7.05 9.19-8.25l11.17-10.4-8.12 12.93c-12.32 19.64-23.46 33.78-33 43-3.51 3.4-6.6 5.9-9.1 7.51a16.43 16.43 0 01-2.61 1.42c-.41.17-.77.27-1.13.3a2.2 2.2 0 01-1.12-.15 2.07 2.07 0 01-1.27-1.91zM511.17 547.4l-2.26 4-1.4-4.38c-3.1-9.83-5.38-24.64-6.01-38-.72-15.2.49-24.32 5.29-24.32 6.74 0 9.83 10.8 10.07 27.05.22 14.28-2.03 29.14-5.7 35.65zm-5.81 58.46l1.53-4.05 2.09 3.8c11.69 21.24 26.86 38.96 43.54 51.31l3.6 2.66-4.39.9c-16.33 3.38-31.54 8.46-52.34 16.85 2.17-.88-21.62 8.86-27.64 11.17l-5.25 2.01 2.8-4.88c12.35-21.5 23.76-47.32 36.05-79.77zm157.62 76.26c-7.86 3.1-24.78.33-54.57-12.39l-7.56-3.22 8.2-.6c23.3-1.73 39.8-.45 49.42 3.07 4.1 1.5 6.83 3.39 8.04 5.55a4.64 4.64 0 01-1.36 6.31 6.7 6.7 0 01-2.17 1.28z" } }] }, name: "file-pdf", theme: "filled" };
function Pe(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Lr(e, a, t[a]);
    });
  }
  return e;
}
function Lr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Ke = function(r, t) {
  var n = Pe({}, r, t.attrs);
  return c(u, Pe({}, n, {
    icon: Vr
  }), null);
};
Ke.displayName = "FilePdfFilled";
Ke.inheritAttrs = !1;
var Hr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM468.53 760v-91.54h59.27c60.57 0 100.2-39.65 100.2-98.12 0-58.22-39.58-98.34-99.98-98.34H424a12 12 0 00-12 12v276a12 12 0 0012 12h32.53a12 12 0 0012-12zm0-139.33h34.9c47.82 0 67.19-12.93 67.19-50.33 0-32.05-18.12-50.12-49.87-50.12h-52.22v100.45z" } }] }, name: "file-ppt", theme: "filled" };
function Ce(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Rr(e, a, t[a]);
    });
  }
  return e;
}
function Rr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var et = function(r, t) {
  var n = Ce({}, r, t.attrs);
  return c(u, Ce({}, n, {
    icon: Hr
  }), null);
};
et.displayName = "FilePptFilled";
et.inheritAttrs = !1;
var kr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM320 482a8 8 0 00-8 8v48a8 8 0 008 8h384a8 8 0 008-8v-48a8 8 0 00-8-8H320zm0 136a8 8 0 00-8 8v48a8 8 0 008 8h184a8 8 0 008-8v-48a8 8 0 00-8-8H320z" } }] }, name: "file-text", theme: "filled" };
function Se(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Br(e, a, t[a]);
    });
  }
  return e;
}
function Br(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var tt = function(r, t) {
  var n = Se({}, r, t.attrs);
  return c(u, Se({}, n, {
    icon: kr
  }), null);
};
tt.displayName = "FileTextFilled";
tt.inheritAttrs = !1;
var Dr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM512 566.1l52.81 197a12 12 0 0011.6 8.9h31.77a12 12 0 0011.6-8.88l74.37-276a12 12 0 00.4-3.12 12 12 0 00-12-12h-35.57a12 12 0 00-11.7 9.31l-45.78 199.1-49.76-199.32A12 12 0 00528.1 472h-32.2a12 12 0 00-11.64 9.1L434.6 680.01 388.5 481.3a12 12 0 00-11.68-9.29h-35.39a12 12 0 00-3.11.41 12 12 0 00-8.47 14.7l74.17 276A12 12 0 00415.6 772h31.99a12 12 0 0011.59-8.9l52.81-197z" } }] }, name: "file-word", theme: "filled" };
function je(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      qr(e, a, t[a]);
    });
  }
  return e;
}
function qr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var rt = function(r, t) {
  var n = je({}, r, t.attrs);
  return c(u, je({}, n, {
    icon: Dr
  }), null);
};
rt.displayName = "FileWordFilled";
rt.inheritAttrs = !1;
var Wr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M854.6 288.7c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2zM296 136v64h64v-64h-64zm64 64v64h64v-64h-64zm-64 64v64h64v-64h-64zm64 64v64h64v-64h-64zm-64 64v64h64v-64h-64zm64 64v64h64v-64h-64zm-64 64v64h64v-64h-64zm0 64v160h128V584H296zm48 48h32v64h-32v-64z" } }] }, name: "file-zip", theme: "filled" };
function xe(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Ur(e, a, t[a]);
    });
  }
  return e;
}
function Ur(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var nt = function(r, t) {
  var n = xe({}, r, t.attrs);
  return c(u, xe({}, n, {
    icon: Wr
  }), null);
};
nt.displayName = "FileZipFilled";
nt.inheritAttrs = !1;
var Zr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z" } }] }, name: "left", theme: "outlined" };
function Fe(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Xr(e, a, t[a]);
    });
  }
  return e;
}
function Xr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Z = function(r, t) {
  var n = Fe({}, r, t.attrs);
  return c(u, Fe({}, n, {
    icon: Zr
  }), null);
};
Z.displayName = "LeftOutlined";
Z.inheritAttrs = !1;
var Yr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z" } }, { tag: "path", attrs: { d: "M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z" } }] }, name: "plus", theme: "outlined" };
function Ae(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Gr(e, a, t[a]);
    });
  }
  return e;
}
function Gr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var X = function(r, t) {
  var n = Ae({}, r, t.attrs);
  return c(u, Ae({}, n, {
    icon: Yr
  }), null);
};
X.displayName = "PlusOutlined";
X.inheritAttrs = !1;
var Qr = { icon: { tag: "svg", attrs: { viewBox: "64 64 896 896", focusable: "false" }, children: [{ tag: "path", attrs: { d: "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z" } }] }, name: "right", theme: "outlined" };
function $e(e) {
  for (var r = 1; r < arguments.length; r++) {
    var t = arguments[r] != null ? Object(arguments[r]) : {}, n = Object.keys(t);
    typeof Object.getOwnPropertySymbols == "function" && (n = n.concat(Object.getOwnPropertySymbols(t).filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    }))), n.forEach(function(a) {
      Jr(e, a, t[a]);
    });
  }
  return e;
}
function Jr(e, r, t) {
  return r in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e;
}
var Y = function(r, t) {
  var n = $e({}, r, t.attrs);
  return c(u, $e({}, n, {
    icon: Qr
  }), null);
};
Y.displayName = "RightOutlined";
Y.inheritAttrs = !1;
const Me = 1, on = /* @__PURE__ */ ze({
  name: "AXAttachmentsFileList",
  __name: "FileList",
  props: {
    prefixCls: {},
    items: {},
    onRemove: {},
    overflow: {},
    upload: {},
    imageProps: {},
    listClassName: {},
    listStyle: {},
    itemClassName: {},
    itemStyle: {}
  },
  setup(e) {
    const r = dt(() => `${e.prefixCls}-list`), t = mt("file-list-container"), [n, a] = I(!1), i = ht();
    gt(() => {
      a(!0), pt(() => {
        a(!1);
      });
    });
    const [o, l] = I(!1), [f, d] = I(!1), b = () => {
      const s = t.value;
      s && (e.overflow === "scrollX" ? (l(Math.abs(s.scrollLeft) >= Me), d(s.scrollWidth - s.clientWidth - Math.abs(s.scrollLeft) >= Me)) : e.overflow === "scrollY" && (l(s.scrollTop !== 0), d(s.scrollHeight - s.clientHeight !== s.scrollTop)));
    };
    bt([() => e.overflow, () => e.items.length], () => {
      Te(() => {
        b();
      });
    }, {
      immediate: !0
    });
    const v = (s) => {
      const w = t.value;
      w && w.scrollTo({
        left: w.scrollLeft + s * w.clientWidth,
        behavior: "smooth"
      });
    }, C = () => {
      v(-1);
    }, P = () => {
      v(1);
    };
    return () => c("div", {
      class: J(r.value, {
        [`${r.value}-overflow-${e.overflow}`]: e.overflow,
        [`${r.value}-overflow-ping-start`]: o.value,
        [`${r.value}-overflow-ping-end`]: f.value
      }, e.listClassName),
      ref: "file-list-container",
      onScroll: b,
      style: e.listStyle
    }, [e.items.map((s) => c(Ot, {
      key: s.uid,
      prefixCls: e.prefixCls,
      item: s,
      onRemove: e.onRemove,
      className: J(e.itemClassName),
      imageProps: e.imageProps,
      style: {
        ...e.itemStyle
      }
    }, null)), !i.value.disabled && c(yt, {
      upload: e.upload,
      children: c(_, {
        class: `${r.value}-upload-btn`,
        type: "dashed"
      }, {
        default: () => [c(X, {
          class: `${r.value}-upload-btn-icon`
        }, null)]
      })
    }, null), e.overflow === "scrollX" && c(vt, null, [c(_, {
      size: "small",
      shape: "circle",
      class: `${r.value}-prev-btn`,
      icon: c(Z, null, null),
      onClick: C
    }, null), c(_, {
      size: "small",
      shape: "circle",
      class: `${r.value}-next-btn`,
      icon: c(Y, null, null),
      onClick: P
    }, null)])]);
  }
});
export {
  De as A,
  Ze as C,
  Ye as E,
  Ge as F,
  Z as L,
  Y as R,
  on as _,
  Qe as a,
  Je as b,
  Ke as c,
  et as d,
  rt as e,
  nt as f,
  tt as g,
  Xe as h,
  Ue as i,
  qe as j,
  We as k
};
