// ant-design-x-vue@1.2.7/es/attachments/FileList/Progress.mjs
import { defineComponent as s, createVNode as e, createTextVNode as n } from "vue";
import { theme as a, Progress as p } from "ant-design-vue";
const l = /* @__PURE__ */ s({
  name: "AXAttachmentsProgress",
  __name: "Progress",
  props: {
    prefixCls: {},
    type: {},
    percent: {},
    format: {},
    status: {},
    showInfo: {
      type: Boolean
    },
    strokeWidth: {},
    strokeLinecap: {},
    strokeColor: {},
    trailColor: {},
    width: {},
    success: {},
    gapDegree: {},
    gapPosition: {},
    size: {},
    steps: {},
    successPercent: {},
    title: {},
    progressStatus: {}
  },
  setup(t) {
    const {
      token: o
    } = a.useToken();
    return () => e(p, {
      type: "circle",
      percent: t.percent,
      size: o.value.fontSizeHeading2 * 2,
      strokeColor: "#FFF",
      trailColor: "rgba(255, 255, 255, 0.3)",
      format: (r) => e("span", {
        style: {
          color: "#FFF"
        }
      }, [(r || 0).toFixed(0), n("%")])
    }, null);
  }
});
export {
  l as default
};
