// ant-design-x-vue@1.2.7/es/components.ts2.mjs
import "./x-provider/index.mjs";
import o from "./bubble/index.mjs";
import "./conversations/index.mjs";
import "./welcome/index.mjs";
import "./prompts/index.mjs";
import "./suggestion/index.mjs";
import r from "./sender/index.mjs";
import m from "./attachments/index.mjs";
import "./thought-chain/index.mjs";
import { _ as t } from "./x-provider/index2.mjs";
import i, { defaultPrefixCls as e } from "./x-provider/hooks/use-x-provider-context.mjs";
import f from "./bubble/BubbleList.mjs";
import p from "./conversations/Conversations.mjs";
import n from "./welcome/Welcome.mjs";
import s from "./prompts/Prompts.mjs";
import _ from "./suggestion/Suggestion.mjs";
import a from "./thought-chain/ThoughtChain.mjs";
const T = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Attachments: m,
  Bubble: o,
  BubbleList: f,
  Conversations: p,
  Prompts: s,
  Sender: r,
  Suggestion: _,
  ThoughtChain: a,
  Welcome: n,
  XProvider: t,
  defaultPrefixCls: e,
  useXProviderContext: i
}, Symbol.toStringTag, { value: "Module" }));
export {
  T as c
};
