// ant-design-x-vue@1.2.7/es/components.mjs
import "./x-provider/index.mjs";
import { default as p } from "./bubble/index.mjs";
import "./conversations/index.mjs";
import "./welcome/index.mjs";
import "./prompts/index.mjs";
import "./suggestion/index.mjs";
import { default as l } from "./sender/index.mjs";
import { default as i } from "./attachments/index.mjs";
import "./thought-chain/index.mjs";
import { _ as n } from "./x-provider/index2.mjs";
import { defaultPrefixCls as h, default as C } from "./x-provider/hooks/use-x-provider-context.mjs";
import { default as g } from "./bubble/BubbleList.mjs";
import { default as c } from "./conversations/Conversations.mjs";
import { default as S } from "./welcome/Welcome.mjs";
import { default as A } from "./prompts/Prompts.mjs";
import { default as T } from "./suggestion/Suggestion.mjs";
import { default as _ } from "./thought-chain/ThoughtChain.mjs";
export {
  i as Attachments,
  p as Bubble,
  g as BubbleList,
  c as Conversations,
  A as Prompts,
  l as Sender,
  T as Suggestion,
  _ as ThoughtChain,
  S as Welcome,
  n as XProvider,
  h as defaultPrefixCls,
  C as useXProviderContext
};
