// ant-design-x-vue@1.2.7/es/index.mjs
import { c as o } from "./components.ts2.mjs";
import { default as l } from "./x-stream/x-stream.mjs";
import { default as d } from "./x-request/x-fetch.mjs";
import { default as p } from "./x-request/x-request.mjs";
import { XAgent as c, default as h } from "./use-x-agent/use-x-agent.mjs";
import { default as b } from "./use-x-chat/use-x-chat.mjs";
import { theme as C } from "ant-design-vue";
import { _ as v } from "./x-provider/index2.mjs";
import { defaultPrefixCls as S, default as B } from "./x-provider/hooks/use-x-provider-context.mjs";
import { default as k } from "./bubble/BubbleList.mjs";
import { default as y } from "./conversations/Conversations.mjs";
import { default as F } from "./welcome/Welcome.mjs";
import { default as O } from "./prompts/Prompts.mjs";
import { default as T } from "./suggestion/Suggestion.mjs";
import { default as _ } from "./thought-chain/ThoughtChain.mjs";
import { default as z } from "./bubble/index.mjs";
import { default as G } from "./sender/index.mjs";
import { default as I } from "./attachments/index.mjs";
const a = function(e) {
  return Object.keys(o).forEach((r) => {
    const t = o[r];
    t.install && e.use(t);
  }), e;
}, s = {
  install: a
};
export {
  I as Attachments,
  z as Bubble,
  k as BubbleList,
  y as Conversations,
  O as Prompts,
  G as Sender,
  T as Suggestion,
  _ as ThoughtChain,
  F as Welcome,
  c as XAgent,
  d as XFetch,
  v as XProvider,
  p as XRequest,
  l as XStream,
  s as default,
  S as defaultPrefixCls,
  a as install,
  C as theme,
  h as useXAgent,
  b as useXChat,
  B as useXProviderContext
};
