// ant-design-x-vue@1.2.7/es/sender/StopLoading.mjs
import { defineComponent as o, createVNode as r, createTextVNode as n } from "vue";
const a = /* @__PURE__ */ o({
  name: "AXSenderStopLoading",
  __name: "StopLoading",
  props: {
    className: null
  },
  setup(t) {
    const e = t;
    return () => r("svg", {
      color: "currentColor",
      viewBox: "0 0 1000 1000",
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      class: e.className
    }, [r("title", null, [n("Stop Loading")]), r("rect", {
      fill: "currentColor",
      height: "250",
      rx: "24",
      ry: "24",
      width: "250",
      x: "375",
      y: "375"
    }, null), r("circle", {
      cx: "500",
      cy: "500",
      fill: "none",
      r: "450",
      stroke: "currentColor",
      "stroke-width": "100",
      opacity: "0.45"
    }, null), r("circle", {
      cx: "500",
      cy: "500",
      fill: "none",
      r: "450",
      stroke: "currentColor",
      "stroke-width": "100",
      "stroke-dasharray": "600 9999999"
    }, [r("animateTransform", {
      attributeName: "transform",
      dur: "1s",
      from: "0 500 500",
      repeatCount: "indefinite",
      to: "360 500 500",
      type: "rotate"
    }, null)])]);
  }
});
export {
  a as default
};
