// ant-design-x-vue@1.2.7/es/sender/components/SendButton.mjs
import { defineComponent as t, mergeDefaults as o, createVNode as l, mergeProps as r } from "vue";
import { a } from "../../attachments/Attachments2.mjs";
import u from "./ActionButton/index.mjs";
import { A as s } from "../../attachments/FileList/FileList2.mjs";
const f = /* @__PURE__ */ t({
  name: "AXSenderLoadingButton",
  __name: "SendButton",
  props: o({
    prefixCls: null,
    type: null,
    htmlType: null,
    shape: null,
    size: null,
    loading: null,
    disabled: null,
    ghost: null,
    block: null,
    danger: null,
    icon: null,
    href: null,
    target: null,
    title: null,
    onClick: null,
    onMousedown: null
  }, {
    type: "primary",
    shape: "circle",
    disabled: void 0
  }),
  setup(e) {
    const n = a(e, ["type", "shape", "disabled"]);
    return () => l(u, r({
      type: e.type,
      shape: e.shape,
      disabled: e.disabled
    }, n, {
      action: "onSend"
    }), {
      default: () => [l(s, null, null)]
    });
  }
});
export {
  f as default
};
